src/qtism/runtime/storage/binary/QtiBinaryStreamAccess.php

```diff
@@ -78,22 +80,30 @@ class QtiBinaryStreamAccess extends BinaryStreamAccess
 
     const RW_CORRECTRESPONSE = 2;
 
-    /**
-     * @var FileManager
-     */
+    private const VARIABLE_TYPES = [
+        'outcomeDeclaration',
+        'responseDeclaration',
+        'templateDeclaration',
+    ];
+
+    /** @var FileManager */
     private $fileManager;
 
+    /** @var VariableFactoryInterface */
+    private $variableFactory;
+
     /**
      * Create a new QtiBinaryStreamAccess object.
      *
      * @param IStream $stream The IStream object to be accessed.
      * @param FileManager $fileManager The FileManager object to handle file variable.
      * @throws StreamAccessException
      */
-    public function __construct(IStream $stream, FileManager $fileManager)
+    public function __construct(IStream $stream, FileManager $fileManager, VariableFactoryInterface $variableFactory)
```

Comment:
This is introducing a dependency for an internal encoding issue, exposing some part of the encoding mechanism. It forces all calls to this class to create and inject a new dependency that should be internal.

This class is used outside of this repository. See:
- https://github.com/oat-sa/bundle-qti/blob/1aca1feb280c82db08538887bb0e008589865463/Accessor/TestSessionAccessor.php#L77
- https://github.com/oat-sa/extension-tao-testqti/blob/f4c615c49373432bfb259c214d73504dce4d2e39/helpers/class.TestSessionStorage.php#L290

Adding a dependency to the constructor is forcing users to change their calls. Please remember this is open source.
At worst, make it optional and instantiate the default implementation in the constructor when it's not provided.
At best, since this factory is just for internal purpose, directly instantiate it or remove the dependency (used only for a mapping lookup).

Reply:
Yeah, I was expecting something like that.
Thanks for pointing to the exact usages!